pkg/cluster/mutate_create_test.go

```diff
@@ -0,0 +1,165 @@
+package cluster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/cluster-api/api/v1alpha3"
+
+	"github.com/giantswarm/azure-admission-controller/pkg/mutator"
+)
+
+func TestClusterCreateMutate(t *testing.T) {
+	type testCase struct {
+		name         string
+		cluster      []byte
+		patches      []mutator.PatchOperation
+		errorMatcher func(err error) bool
+	}
+
+	clusterNetwork := &v1alpha3.ClusterNetwork{
+		APIServerPort: to.Int32Ptr(443),
+		ServiceDomain: "ab123.k8s.test.westeurope.azure.gigantic.io",
+		Services: &v1alpha3.NetworkRanges{
+			CIDRBlocks: []string{
+				"172.31.0.0/16",
+			},
+		},
+	}
+
+	testCases := []testCase{
+		{
+			name:    fmt.Sprintf("case 0: ControlPlaneEndpoint left empty"),
+			cluster: clusterRawObject("ab132", clusterNetwork, "", 0),
+			patches: []mutator.PatchOperation{
+				{
+					Operation: "add",
+					Path:      "/spec/controlPlaneEndpoint/host",
+					Value:     "api.ab132.k8s.test.westeurope.azure.gigantic.io",
+				},
+				{
+					Operation: "add",
+					Path:      "/spec/controlPlaneEndpoint/port",
+					Value:     443,
+				},
+			},
+			errorMatcher: nil,
+		},
+		{
+			name:         fmt.Sprintf("case 1: ControlPlaneEndpoint has a value"),
+			cluster:      clusterRawObject("ab132", clusterNetwork, "api.giantswarm.io", 123),
+			patches:      []mutator.PatchOperation{},
+			errorMatcher: nil,
+		},
+		// This test doesn't work because the clusterNetwork struct has pointers in it and we can't compare them.
```

Comment:
So let's remove it? :smile:

Reply:
having that test would be useful, but we can't implement it.
I'd like to keep track of the need of having it though.
Is this an hard blocker for you?

Reply:
not at all, let's keep it then